omnibus/config/templates/openssl-fips/fipsinstall.sh.erb

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# The OpenSSL security policy states:
+# "The Module shall have the self-tests run, and the Module config file output generated on each
+# platform where it is intended to be used. The Module config file output data shall not be copied from
+# one machine to another."
+# This script aims to run self-tests and generate `fipsmodule.cnf.`
+# Because the provided `openssl.cnf` references to `fipsmodule.cnf` which is not yet created, we first create it
+# as `openssl.cnf.tmp` and then move it to its final name `openssl.cnf` when `fipsmodule.cnf` has been created
+
+set -euo pipefail
+
+INSTALL_DIR="<%= install_dir %>/embedded"
+
+FIPS_MODULE_PATH="${INSTALL_DIR}/ssl/fipsmodule.cnf"
+OPENSSL_CONF_PATH="${INSTALL_DIR}/ssl/openssl.cnf"
+
+FIPS_SO_PATH="${INSTALL_DIR}/lib/ossl-modules/fips.so"
+OPENSSL_BIN="${INSTALL_DIR}/bin/openssl"
+
+
+if [ ! -f "${FIPS_MODULE_PATH}" ]; then
+    "${OPENSSL_BIN}" fipsinstall -module "${FIPS_SO_PATH}" -out "${FIPS_MODULE_PATH}"
+    mv "${OPENSSL_CONF_PATH}.tmp" "${OPENSSL_CONF_PATH}"
+fi
+
+if ! "${OPENSSL_BIN}" fipsinstall -module "${FIPS_SO_PATH}" -in "${FIPS_MODULE_PATH}" -verify; then
+    echo "openssl fipsinstall: verification of FIPS compliance failed. $INSTALL_DIR/fipsmodule.cnf was corrupted or the installation failed."
```

Comment:
Shouldn't we fail harder here, or at least propagate the failure to the package script?

Reply:
We can, the Agent won't be able to start with GOFIPS=1 in any case. Do we want to impact non-FIPS customers if FIPS cannot be enabled ? (this should not happen but we never know)

Reply:
Ah indeed there's nothing FIPS specific during the install, so maybe not a hard failure.

However it would be better to display this in `stderr`, I'm not entirely sure package managers forward `stdout` to the users' terminal